spring-jdbc/src/main/java/org/springframework/jdbc/datasource/init/ResourceDatabasePopulator.java

```diff
@@ -171,9 +171,22 @@ public void setSeparator(String separator) {
 	 * Set the prefix that identifies single-line comments within the SQL scripts.
 	 * <p>Defaults to {@code "--"}.
 	 * @param commentPrefix the prefix for single-line comments
+	 * @see #setCommentPrefixes(String...)
 	 */
 	public void setCommentPrefix(String commentPrefix) {
-		this.commentPrefix = commentPrefix;
+		Assert.hasText(commentPrefix, "CommentPrefix must not be null or empty");
```

Comment:
```suggestion
		Assert.notNull(commentPrefixes, "'commentPrefix' must not be null or empty");
```

Reply:
We seem to have mixed styles in the Framework codebase. Other asserts in `ResourceDatabasePopulator` use the form I picked. See `setBlockCommentStartDelimiter` for an example. I personally prefer that style but I didn't want to change ScriptUtils.

Reply:
> We seem to have mixed styles in the Framework codebase. 

Yes, that's unfortunately true. <shrug>